rn/Teacher/ios/Submissions/HideGradesViewController.swift

```diff
@@ -0,0 +1,108 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import UIKit
+import Core
+
+class HideGradesViewController: UIViewController {
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var hideGradesButton: DynamicButton!
+    private var sections: [String] = []
+    private var sectionToggles: [Bool] = []
+    private var gradesCurrentlyPosted = 0
+    private var showSections = false
+    @IBOutlet weak var allGradesHiddenView: UIView!
+    @IBOutlet weak var allHiddenLabel: DynamicLabel!
+    @IBOutlet weak var allHiddenSubHeader: DynamicLabel!
+
+    static func create() -> HideGradesViewController {
+        let controller = loadFromStoryboard()
+        return controller
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setupSections()
+
+        allGradesHiddenView.isHidden = true
+        allHiddenLabel.text = NSLocalizedString("All Hidden", comment: "")
+        allHiddenSubHeader.text = NSLocalizedString("All grades are currently hidden.", comment: "")
+
+        hideGradesButton.setTitle(NSLocalizedString("Hide Grades", comment: ""), for: .normal)
+    }
+
+    func setupTableView() {
+        tableView.register(PostGradesViewController.SectionCell.self, forCellReuseIdentifier: String(describing: PostGradesViewController.SectionCell.self))
```

Comment:
nitpick: you can omit the second argument, right?

Reply:
fixed